include/swift/Syntax/Trivia.h.gyb

```diff
@@ -410,4 +412,56 @@ struct ScalarEnumerationTraits<syntax::TriviaKind> {
 };
 } // namespace json
 } // namespace swift
+
+namespace llvm {
+namespace yaml {
+
+/// Deserialization traits for TriviaPiece.
+/// - All trivia pieces will have a "kind" key that contains the serialized
+///   name of the trivia kind.
+/// - Comment trivia will have the associated text of the comment under the
+///   "value" key.
+/// - All other trivia will have the associated integer count of their
+///   occurrences under the "value" key.
+template<>
+struct MappingTraits<swift::syntax::TriviaPiece> {
+  static swift::syntax::TriviaPiece mapping(IO &in) {
+    swift::syntax::TriviaKind kind;
+    in.mapRequired("kind", kind);
+    switch (kind) {
+% for trivia in TRIVIAS:
+    case swift::syntax::TriviaKind::${trivia.name}: {
+% if trivia.is_collection():
+
+/// It would raise error when deserializing number with trailing character
+/// eg. "1\n"
```

Comment:
Has this happened while testing this? If so, then the serialization should probably be updated.

Reply:
Yes, it happened. I'll take more tests to figure out this problem since the llvm parser should tokenize only `"1"` instead of `"1\n"`.

Reply:
Oh, that's fine. Can you link this bug in a FIXME comment above the workaround?